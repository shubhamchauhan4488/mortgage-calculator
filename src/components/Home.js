import React, { useState } from "react";
import { MortgageForm } from "./MortgageForm";

export const Home = () => {
  const [mortgageAmt, setMortgageAmt] = useState(50000);
  const [rate, setRate] = useState(5);
  const [amortizationPeriod, setAmortizationPeriod] = useState(25);
  const [paymentFrequency, setPaymentFrequency] = useState(12);
  const [term, setTerm] = useState(5);
  const [errors, setErrors] = useState({
    mortgageAmt: "",
    rate: "",
    amortizationPeriod: "",
    paymentFrequency: "",
    term: "",
  });

  const handleInputChange = (e) => {
    setErrors({
      ...errors,
      [e.target.name]: "",
    });
    switch (e.target.name) {
      case "mortgageAmt": {
        setMortgageAmt(e.target.value);
        break;
      }
      case "rate": {
        setRate(e.target.value);
        break;
      }
      case "amortizationPeriod": {
        setAmortizationPeriod(e.target.value);
        break;
      }
      case "paymentFrequency": {
        setPaymentFrequency(e.target.value);
        break;
      }
      case "term": {
        setTerm(e.target.value);
        break;
      }
      default:
        return;
    }
  };

  return (
    <div className="main-content-container">
      <MortgageForm
        mortgageAmt={mortgageAmt}
        rate={rate}
        amortizationPeriod={amortizationPeriod}
        paymentFrequency={paymentFrequency}
        term={term}
        handleInputChange={handleInputChange}
        errors={errors}
      />
    </div>
  );
};
