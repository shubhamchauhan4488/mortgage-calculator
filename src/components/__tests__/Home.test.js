import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Home } from '../Home';

describe('Home', () => {
  afterEach(cleanup);

  it('renders correctly', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders Mortgage form child correctly', () => {
    const { getByTestId } = render(<Home />);
    expect(getByTestId('mortgage-form-container')).toBeInTheDocument();
  });
});
