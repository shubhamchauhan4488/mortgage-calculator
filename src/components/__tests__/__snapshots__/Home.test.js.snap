// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<DocumentFragment>
  <div
    class="main-container"
  >
    <div
      class="jumbotron no-gutters col-12 col-sm-6 mt-3 py-5"
      data-testid="mortgage-form-container"
      id="mortgage-form-container"
    >
      <div
        class="row align-items-center mb-3"
      >
        <div
          class="col-12 col-sm-6"
        >
          <label
            for="basic-url"
          >
            Mortgage Amount
          </label>
        </div>
        <div
          class="col-12 col-sm-6"
        >
          <div
            class="input-group"
          >
            <div
              class="input-group-prepend"
            >
              <span
                class="input-group-text"
              >
                $
              </span>
            </div>
            <input
              aria-label="Amount (to the nearest dollar)"
              class="form-control"
              data-testid="mortgage-amt"
              id="mortgage-amt"
              name="mortgageAmt"
              placeholder="Value nearest to dollar amount"
              type="number"
              value="50000"
            />
            <div
              class="input-group-append"
            >
              <span
                class="input-group-text"
              >
                .00
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row align-items-center mb-3 align-items-center mb-3"
      >
        <div
          class="col-12 col-sm-6"
        >
          <label
            for="interestRate"
          >
            Interest Rate
          </label>
        </div>
        <div
          class="col-12 col-sm-6"
        >
          <div
            class="input-group"
          >
            <div
              class="input-group-prepend"
            >
              <span
                class="input-group-text"
              >
                %
              </span>
            </div>
            <input
              aria-label="Interest Rate"
              class="form-control"
              data-testid="rate"
              id="rate"
              max="100"
              min="0"
              name="rate"
              placeholder="Annual Interest Rate"
              type="number"
              value="5"
            />
          </div>
        </div>
      </div>
      <div
        class="row align-items-center mb-3"
      >
        <div
          class="col-12 col-sm-6"
        >
          <label
            for="amortizationPeriod"
          >
            Amortization Period
          </label>
        </div>
        <div
          class="col-12 col-sm-6"
        >
          <input
            aria-label="Amortization Period"
            class="form-control"
            data-testid="amortization-period"
            id="amortization-period"
            max="30"
            min="1"
            name="amortizationPeriod"
            placeholder="Years"
            type="number"
            value="25"
          />
        </div>
      </div>
      <div
        class="row align-items-center mb-3"
      >
        <div
          class="col-12 col-sm-6"
        >
          <label
            for="paymentFrequency"
          >
            Payment Frequency
          </label>
        </div>
        <div
          class="col-12 col-sm-6"
        >
          <select
            data-testid="payment-frequency"
            id="payment-frequency"
            name="paymentFrequency"
          >
            <option
              id="0"
              value="52"
            >
              Weekly
            </option>
            <option
              id="1"
              value="26"
            >
              Bi-Weekly (every 2 weeks)
            </option>
            <option
              id="2"
              value="12"
            >
              Monthly (12x per year)
            </option>
            <option
              id="3"
              value="24"
            >
              Semi-monthly (24x per year)
            </option>
          </select>
        </div>
      </div>
      <div
        class="row align-items-center mb-3"
      >
        <div
          class="col-12 col-sm-6"
        >
          <label
            for="term"
          >
            Term
          </label>
        </div>
        <div
          class="col-12 col-sm-6"
        >
          <select
            data-testid="term"
            id="term"
            name="term"
          >
            <option
              id="0"
              value="1"
            >
              1
            </option>
            <option
              id="1"
              value="2"
            >
              2
            </option>
            <option
              id="2"
              value="3"
            >
              3
            </option>
            <option
              id="3"
              value="4"
            >
              4
            </option>
            <option
              id="4"
              value="5"
            >
              5
            </option>
            <option
              id="5"
              value="6"
            >
              6
            </option>
            <option
              id="6"
              value="7"
            >
              7
            </option>
            <option
              id="7"
              value="8"
            >
              8
            </option>
            <option
              id="8"
              value="9"
            >
              9
            </option>
            <option
              id="9"
              value="10"
            >
              10
            </option>
            <option
              id="10"
              value="11"
            >
              11
            </option>
            <option
              id="11"
              value="12"
            >
              12
            </option>
            <option
              id="12"
              value="13"
            >
              13
            </option>
            <option
              id="13"
              value="14"
            >
              14
            </option>
            <option
              id="14"
              value="15"
            >
              15
            </option>
            <option
              id="15"
              value="16"
            >
              16
            </option>
            <option
              id="16"
              value="17"
            >
              17
            </option>
            <option
              id="17"
              value="18"
            >
              18
            </option>
            <option
              id="18"
              value="19"
            >
              19
            </option>
            <option
              id="19"
              value="20"
            >
              20
            </option>
            <option
              id="20"
              value="21"
            >
              21
            </option>
            <option
              id="21"
              value="22"
            >
              22
            </option>
            <option
              id="22"
              value="23"
            >
              23
            </option>
            <option
              id="23"
              value="24"
            >
              24
            </option>
            <option
              id="24"
              value="25"
            >
              25
            </option>
            <option
              id="25"
              value="26"
            >
              26
            </option>
            <option
              id="26"
              value="27"
            >
              27
            </option>
            <option
              id="27"
              value="28"
            >
              28
            </option>
            <option
              id="28"
              value="29"
            >
              29
            </option>
            <option
              id="29"
              value="30"
            >
              30
            </option>
          </select>
        </div>
      </div>
      <div
        class="row justify-content-center"
      >
        <button
          class="btn btn-dark"
          data-testid="calculate-btn"
          id="calculate-btn"
          type="button"
        >
          Calculate
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
